reinforcement_learning/rlclientlib/logger/async_batcher.h

```diff
@@ -63,8 +65,14 @@ namespace reinforcement_learning {
 
   template<typename TEvent>
   int async_batcher<TEvent>::append(TEvent&& evt, api_status* status) {
+    if (_queue_dropping_disable && _queue.size() > _queue_max_size) { // block if queue is full
+      std::unique_lock<std::mutex> lk(_m);
+      _cv.wait(lk, [this] { return _queue.size() < _queue_max_size; });
+    }
     _queue.push(std::move(evt));
-    prune_if_needed();
+    if (!_queue_dropping_disable && _queue.size() > _queue_max_size) { // drop events if queue is full
```

Comment:
why do we need to split this logic into 2 blocks?
if (queue is full)
  if (drop) drop else wait;
looks cleaner.

Reply:
Because:
* the prunning is done after the enqueuing, whereas the waiting must happen before. 
* I tried to avoid modifying the prunning logic.

If you think that the prunning logic can be safely moved from "after the enqueuing" to "before the enqueing", then yes, the logic does not need to be splitted.

Reply:
I do not see big difference for moving any of blocks either before or after push, since effectively it is just shifting _queue_max_size by 1 (which is not a problem for me).

Reply:
+1